internal/resources/resource_host.go

```diff
@@ -431,6 +454,37 @@ func resourceMetalHostCreate(d *schema.ResourceData, meta interface{}) (err erro
 	if err != nil {
 		return err
 	}
+
+	// host create is asynchronous in Metal svc. Wait until host state is Ready.
+	createStateConf := &resource.StateChangeConf{
+		Pending: []string{
+			hStateNew,
+			hStateImagingPrep,
+			hStateImaging,
+			hStateConnecting,
+			hStateBooting,
+		},
+		Target: []string{
+			hStateReady,
```

Comment:
should we handle an error state too and exit the waiting - seems like we should, here and in update

Reply:
Regarding the Pending list, if a new host state (nonfailing one) gets introduced, then it becomes an absolute necessity to use the latest plugin, and with the old plugin, Host create will always fail. 
Don't know how often the 'state' needs updates to support on going enhancements, depending on that, checking only for failed states looks more stable solution.

Reply:
this is how StateChangeConf works; there is no provision to fail on a condition (state)